src/themes/teams/components/Icon/iconVariables.ts

```diff
@@ -1,31 +1,38 @@
+import { ColorValues } from '../../../types'
+import { mapColorsToScheme } from '../../../../lib'
 import { pxToRem } from '../../utils'
 
-export interface IconVariables {
-  [key: string]: string | number | boolean | undefined
```

Comment:
why this extensibility point was removed? Note that this extensibility is necessary for some scenarios, like, for example, to provide some category to the component that will define its styles:

```jsx
<Icon variables={ chatIcon: true } ... />
```

With these changes types will be broken.

Reply:
the reasons were:
### 1. it doesn't bring any value
Your example would work without any type errors because the type for `variables` prop is not generic enough anyway to take into account `IconVariables` interface.

`IconVariables` is only relevant for 
src/themes/teams/components/Icon/iconVariables.ts

### 2. It was causing the following type error:
![screenshot 2018-12-20 at 16 58 32](https://user-images.githubusercontent.com/5442794/50292321-714df880-0479-11e9-8025-65c68fa6d2f7.png)

If you feel strongly that we should add the extensibility back (which I doubt), the error can be fixed by adding `object` to the values type:
![screenshot 2018-12-20 at 16 59 28](https://user-images.githubusercontent.com/5442794/50292369-96426b80-0479-11e9-86dd-5646e65b0f48.png)

@kuzhelov what do you think?

Reply:
my thoughts have the following motivation points
- we would like to improve our typings, so for variables we might expect typings being introduced
- in case if we will remove extensibility point from variables type, we will
  - break variables' semantics by this move, as variables interface won't be extendable anymore 

If we would like to make this move (note, this is not related to this PR), we should make it for all variables' types altogether, to enforce consistency in our code base. However, given our current vision about the fact that variables are seen to be extendable ones, we shouldn't introduce any changes at least for now

Reply:
that's fine with me if it moves the PR closer to a merge, reverted the change as advised